metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/dsl/functions/RestFunctions.java

```diff
@@ -0,0 +1,351 @@
+          name = "GET",
+          description = "Performs a REST GET request and parses the JSON results into a map.",
+          params = {
+                  "url - URI to the REST service",
+                  "rest_config - Optional - Map (in curly braces) of name:value pairs, each overriding the global config parameter " +
+                          "of the same name. Default is the empty Map, meaning no overrides."
+          },
+          returns = "JSON results as a Map")
+  public static class RestGet implements StellarFunction {
+
+    /**
+     * Whether the function has been initialized.
+     */
+    private boolean initialized = false;
+
+    /**
+     * The CloseableHttpClient.
+     */
+    private CloseableHttpClient httpClient;
+
+    /**
+     * Executor used to impose a hard request timeout.
+     */
+    private ScheduledExecutorService scheduledExecutorService;
+
+    /**
+     * Apply the function.
+     * @param args The function arguments including uri and rest config.
+     * @param context Stellar context
+     */
+    @Override
+    public Object apply(List<Object> args, Context context) throws ParseException {
+      RestConfig restConfig = new RestConfig();
+      try {
+        URI uri = new URI(getArg(0, String.class, args));
+        Optional<Object> globalCapability = context.getCapability(GLOBAL_CONFIG, false);
+
+        Map<String, Object> globalConfig = (Map<String, Object>) globalCapability.get();
+
+        restConfig = getRestConfig(args, globalConfig);
+
+        HttpHost target = new HttpHost(uri.getHost(), uri.getPort(), uri.getScheme());
+        Optional<HttpHost> proxy = getProxy(restConfig);
+        HttpClientContext httpClientContext = getHttpClientContext(restConfig, target, proxy);
+
+        HttpGet httpGet = new HttpGet(uri);
+        httpGet.addHeader("Accept", "application/json");
+        httpGet.setConfig(getRequestConfig(restConfig, proxy));
+
+        return doGet(restConfig, httpGet, httpClientContext);
+      } catch (URISyntaxException e) {
+        throw new IllegalArgumentException(e.getMessage(), e);
+      } catch (IOException e) {
+        LOG.error(e.getMessage(), e);
+        return restConfig.getErrorValueOverride();
+      }
+    }
+
+    /**
+     * Perform the HttpClient get and handle the results.  A configurable list of status codes are accepted and the
+     * response content (expected to be json) is parsed into a Map.  Values returned on errors and when response content
+     * is also configurable.  The rest config "timeout" setting is imposed in this method and will abort the get request
+     * if exceeded.
+     *
+     * @param restConfig
+     * @param httpGet
+     * @param httpClientContext
+     * @return
+     * @throws IOException
+     */
+    private Object doGet(RestConfig restConfig, HttpGet httpGet, HttpClientContext httpClientContext) throws IOException {
+
+      // Schedule a command to abort the httpGet request if the timeout is exceeded
```

Comment:
Could you elaborate in the docs somewhere that this is the hard timeout, and the reason we need the hard timeouts is that the lib provided timeouts are insufficient to achieve the hard timeout?

Reply:
Did you see the REST section in the Stellar README?  I can add more content to make that clearer.

Reply:
I may have forgotten to refresh since before the readme was added. I'll check it out.

Reply:
In that README, could you alter
`timeout - Hard timeout for the total request time.  Defaults to 1000 ms.` to be a bit more explicit about it. E.g. something like 
`timeout - Stellar enforced hard timeout for the total request time.  Defaults to 1000 ms. Client timeouts alone are insufficient to guarantee the hard timeout`

Is this a reasonable statement of the problem?  Unless I'm mistaken, this gets alluded to a bit, but is never explicitly stated.

Reply:
I like it.  Done.